test/models/operatingsystem_test.rb

```diff
@@ -56,6 +56,18 @@ class OperatingsystemTest < ActiveSupport::TestCase
     assert operating_system.to_s == operating_system.to_label
   end
 
+  test "should find Debian by fullname string" do
+    str = "Debian 12"
+    os = Operatingsystem.find_by_to_label(str)
+    assert_equal str, os.fullname
+  end
+
+  test "should find Ubuntu by fullname string" do
+    str = "Ubuntu 22.04.3"
+    os = Operatingsystem.find_by_to_label(str)
+    assert_equal str, os.fullname
+  end
```

Comment:
Please use factories. Define if needed in `test/factories/operatingsystem.rb` but given how specific this is to test parsing of version numbers I think defining it locally makes sense as well.